controllers/userController.js

```diff
@@ -0,0 +1,65 @@
+const User = require("../models/userModel");
+const catchAsync = require("../utils/catchAsync");
+const jwt = require("jsonwebtoken");
+const AppError = require("../utils/appError");
+const { validationResult } = require("express-validator");
+
+const signToken = (id) =>
+  jwt.sign(
+    {
+      id,
+    },
+    process.env.JWT_SECRET,
+    {
+      expiresIn: process.env.JWT_EXPIRES_IN,
+    }
+  );
+
+exports.signup = catchAsync(async (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+  const newUser = await User.create({
```

Comment:
Why not check if user already exist before you create?

Reply:
alright, will do that